src/doc/tarpl/vec-layout.md

```diff
@@ -0,0 +1,97 @@
+% Layout
+
+First off, we need to come up with the struct layout. Naively we want this
+design:
```

Comment:
Why? 

That is, it seems like giving some context for what `ptr`, `cap` and `len` are would be nice?